server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/MultiTsFileDeviceIterator.java

```diff
@@ -72,6 +74,23 @@ public MultiTsFileDeviceIterator(List<TsFileResource> tsFileResources) throws IO
     }
   }
 
+  /** Used for cross space compaction. */
+  public MultiTsFileDeviceIterator(
```

Comment:
If exception occurs here, end query may close some files which have not yet build reader.

Reply:
If the reader is closed in this method, there will be problems. Query f1, merge f1 and f2, and throw an exception when creating f2 reader in compaction. If f1 is closed in this method, the query will report an error. And f2 reader will be closed in end query.

Reply:
Get it